astropy/coordinates/baseframe.py

```diff
@@ -709,8 +856,17 @@ def transform_to(self, new_frame):
         if self._data is None:
             raise ValueError('Cannot transform a frame with no data')
 
+        if (getattr(self.data, 'differentials', None) and
+           hasattr(self, 'obstime') and hasattr(new_frame, 'obstime') and
+           np.all(self.obstime == new_frame.obstime)):
```

Comment:
I take back that last part... I was getting those confused with the `FK5` which have a specified equinox.  So yes, it *does* work, and I'm adding a test for that.